serverless.yml

```diff
@@ -24,10 +24,55 @@ provider:
       - "sqs:SendMessage"
      Resource:
       Fn::GetAtt: [ SQSQueue, Arn ]
+   - Effect: Allow
+     Action:
+      - "sqs:SendMessage"
+     Resource:
+      - "Fn::GetAtt":
+        - ReceiverDeadLetterQueue
+        - Arn
   environment:
     HTTPOBS_API_URL: 'https://http-observatory.security.mozilla.org/api/v1'
+    SQS_URL:
+      Ref: SQSQueue
+
+# Adding some packaging information here to clean up
+# and to include the nmap static library
+package:
+  include:
+    - bin/*
+  exclude:
+    - .venv/**
+    - .git/**
+    - __pycache__/**
+    - node_modules/**
 
 functions:
+  onDemandPortScan:
+    handler: handler.addPortScanToQueue
+    events:
+      - http:
+          path: ondemand/portscan
+          method: PUT
+          cors: true
+  cronPortScan:
+    handler: handler.runScheduledPortScan
+    # Giving 5 minutes to run the TCP scan before it times outs
+    timeout: 300
+    events:
+      # Invoke Lambda function once a day
+      # We can fine grain it with cron() later
+      - schedule: 
+          rate: rate(1 day)
```

Comment:
Do we have any control or preference on when the scan happens or is it completely  up to Amazon  to decide when?

Reply:
With the `rate()` syntax, no. But with the `cron()` syntax, yes. Will change it to the cron() syntax.

Reply:
Updated to `cron(0 16 * * ? *)` => 4 PM UTC every day. Note that this function is currently disabled. Let me know if you have a preferred time/frequency.